src/api/api.c

```diff
@@ -837,6 +837,20 @@ const z_loaned_encoding_t *z_reply_err_encoding(const z_loaned_reply_err_t *repl
 const char *z_string_data(const z_loaned_string_t *str) { return str->val; }
 size_t z_string_len(const z_loaned_string_t *str) { return str->len; }
 
+int8_t z_string_from_str(z_owned_string_t *str, const char *value) {
+    str->_val = _z_string_make(value);
+    return _Z_RES_OK;
+}
+
+void z_string_empty(z_owned_string_t *str) { str->_val = _z_string_null(); }
+
+int8_t z_string_from_substr(z_owned_string_t *str, const char *value, size_t len) {
+    str->_val = _z_string_n_make(value, len);
```

Comment:
it might be unsafe to call _z_string_n_make with current implementation if you only allowed to touch len characters, since it will copy len+1 and thus may technically result in segfault.